mods/cnc/maps/nod02a/nod02a.lua

```diff
@@ -0,0 +1,262 @@
+        end)
+        if #globalActors < count then
+            maxUnits = #globalActors
+        else
+            maxUnits = count
+        end        
+		for i = 1, maxUnits, 1 do	
+	        actors[counter] = globalActors[i]	
+            counter = counter + 1		    
+	    end
+    end
+	return actors
+end
+
+OnAnyDamaged = function(actors, func)
+    for key in pairs(actors) do
+        Trigger.OnDamaged(actors[key], func)
+    end
+end
+
+CheckForBase = function(player, buildings)
+	local baseBuildings = Map.ActorsInBox(Map.TopLeft, Map.BottomRight, function(actor)
+        local found = false
+        for i = 1, #buildings, 1 do
+            if actor.Type == buildings[i] then
+                found = true
+            end
+        end
+		return actor.Owner == BadGuy and found
+	end)
+	return #baseBuildings >= 3
+end
+
+Atk7TriggerFunction = function()      
+    MyActors = getActors(GoodGuy, { ['e1'] = 3 })   
+    for key in pairs(MyActors) do
+        Atk4Movement(MyActors[key])
+    end 
+end
+
+Atk7TriggerFunctionTime = 50
+
+DfndTriggerFunction = function()		
+	local list = GoodGuy.GetGroundAttackers()	
+	Utils.Do(list, function(unit)
+		unit.Hunt()
+	end)
+end
+
+Atk6TriggerFunction = function()
+    MyActors = getActors(GoodGuy, { ['e1'] = 4 })       
+    for key in pairs(MyActors) do	    
+        Atk3Movement(MyActors[key])
+    end 
+end
+
+Atk6TriggerFunctionTime = 80
+
+Pat1Movement = function(unit)
+    unit.Move(waypoint0.Location)
+	unit.Move(waypoint1.Location)
+	unit.Move(waypoint2.Location)
+    unit.Move(waypoint3.Location)	
+end
+
+Pat1TriggerFunction = function()
+    MyActors = getActors(GoodGuy, { ['e1'] = 3 })   
+    for key in pairs(MyActors) do
+        Pat1Movement(MyActors[key])
+    end 
+end
+
+Pat1TriggerFunctionTime = 30
+
+Atk1Movement = function(unit)
+    unit.Move(waypoint2.Location)
+	unit.Move(waypoint4.Location)
+	unit.Move(waypoint5.Location)
+    unit.AttackMove(waypoint6.Location)	
+end
+
+Atk2Movement = function(unit)
+    unit.Move(waypoint2.Location)
+	unit.Move(waypoint5.Location)
+	unit.Move(waypoint7.Location)
+    unit.AttackMove(waypoint6.Location)	
+end
+
+Atk2TriggerFunction = function()
+    MyActors = getActors(GoodGuy, { ['e1'] = 3 })   
+    for key in pairs(MyActors) do
+        Atk2Movement(MyActors[key])
+    end 
+end
+
+Atk2TriggerFunctionTime = 40
+
+Atk3Movement = function(unit)
+    unit.Move(waypoint2.Location)
+	unit.Move(waypoint7.Location)
+	unit.Move(waypoint8.Location)
+    unit.AttackMove(waypoint9.Location)	
+end
+
+Atk3TriggerSwitch = false
+Atk3TriggerFunction = function()
+    if Atk3TriggerSwitch == false then
+        Atk3TriggerSwitch = true		
+    	MyActors = getActors(GoodGuy, { ['e1'] = 4 })   
+        for key in pairs(MyActors) do			
+            Atk3Movement(MyActors[key])
+        end 
+	end
+end
+
+Atk4Movement = function(unit)
+    unit.Move(waypoint0.Location)
+	unit.Move(waypoint8.Location)
```

Comment:
The mixing of tabs and spaces makes this a little hard to read. Set your Lua editor to use tabs for indention.